### PR TITLE
Add Peer Auth Translation

### PR DESCRIPTION
**Pull request by mlholland**
Also removes default detination rule translation. 
BOT NOTES: 
resolves https://github.com/solo-io/gloo-mesh-enterprise/issues/833

### CODE REVIEW

File: pkg/mesh-networking/apply/applier.go [hunk 1]
```diff
@@ -447,18 +447,20 @@ func validateAndReturnVirtualMesh(
...
 	// so locking should not be necessary.
-	unappliedTrafficPolicies map[string]map[string][]error // sets.Key(*discoveryv1.Destination)
-	unappliedAccessPolicies  map[string]map[string][]error // sets.Key(*discoveryv1.Destination)
-	unappliedFederations     map[string]map[string][]error // sets.Key(*discoveryv1.Destination)
-	unappliedVirtualMeshes   map[string]map[string][]error // sets.Key(*discoveryv1.Mesh)
+	unappliedTrafficPolicies     map[string]map[string][]error // sets.Key(*discoveryv1.Destination)
+	unappliedAccessPolicies      map[string]map[string][]error // sets.Key(*discoveryv1.Destination)
+	unappliedFederations         map[string]map[string][]error // sets.Key(*discoveryv1.Destination)
+	unappliedVirtualMeshes       map[string]map[string][]error // sets.Key(*discoveryv1.Mesh)
+	unappliedPeerAuthentications map[string]map[string][]error // sets.Key(*discoveryv1.Mesh)
```
Comment (harveyxia) [inline pkg/mesh-networking/apply/applier.go:454]: We can only report errors to GM resources, not external service mesh resources. Also, the input to the translated PeerAuthentication is the Settings object.

File: pkg/mesh-networking/translation/istio/mesh/istio_mesh_translator.go [hunk 2]
```diff
@@ -75,5 +74,7 @@ func (t *translator) Translate(
 		t.mtlsTranslator.Translate(mesh, appliedVirtualMesh, istioOutputs, localOutputs, reporter)
 		t.federationTranslator.Translate(in, mesh, appliedVirtualMesh, istioOutputs, reporter)
 		t.accessTranslator.Translate(mesh, appliedVirtualMesh, istioOutputs)
+	} else {
+		contextutils.LoggerFrom(t.ctx).Debugf("skipping translation elements that require an applied virtual mesh")
```
Comment (harveyxia) [inline pkg/mesh-networking/translation/istio/mesh/istio_mesh_translator.go:78]: we don't need this log line

File: pkg/mesh-networking/translation/istio/destination/destinationrule/destination_rule_translator.go [hunk 3]
```diff
@@ -212,15 +210,6 @@ func (t *translator) initializeDestinationRule(
 		},
 	}
```
Comment (harveyxia) [inline pkg/mesh-networking/translation/istio/destination/destinationrule/destination_rule_translator.go:212]: if the destinationRule has no effect, don't output to snapshot

File: pkg/mesh-networking/translation/istio/mesh/peer-auth/peer_auth_translator.go [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+) (*security_istio_io_v1beta1.PeerAuthentication, error) {
+	istioTlsMode, err := tls.MapIstioTlsModeToPeerAuth(t.settings.Spec.GetPeerAuth().GetPeerAuthTlsMode())
+
+	if err != nil {
+		return nil, err
+	}
+	// note: if 'mode' is set to the UNSET enum, then the resulting yaml will simply not set this value.
+	// Which I found unintuitive, but it does make literal sense.
+	newPeerAuth := &security_istio_io_v1beta1.PeerAuthentication{
+		ObjectMeta: metav1.ObjectMeta{
```
Comment (harveyxia) [inline pkg/mesh-networking/translation/istio/mesh/peer-auth/peer_auth_translator.go:91]: 1. You need to include the [translated object labels](https://github.com/solo-io/gloo-mesh/blob/ec306c50b8c7c366a708078982cea7b1224b7210/pkg/mesh-networking/translation/utils/metautils/object_meta.go#L78)

2. Not sure why we're using Agent namespace, instead the namespace should be the istio root namespace, `mesh.Spec.GetIstio().GetInstallation().GetNamespace()`

File: pkg/mesh-networking/apply/applier.go (hunk 1 above)
Comment (mlholland) [reply pkg/mesh-networking/apply/applier.go:465]: I don't think I understand what the resulting applier should look like. Can we discuss this when you return?

**Comment by mlholland** [comment]
According to @EItanya , the OSM tests should be removed. I _think_ I did that, but github actions doesn't seem to have caught the message. Ignoring the two permanently expected OSM tests, this is passing CI.

**Pull request closed without merging**